projects/js-packages/publicize-components/src/hooks/use-media-restrictions/restrictions.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * These restrictions were updated on: November 18, 2022.
+ *
+ * Image and video size is in MB.
+ * Video length is in seconds.
+ */
+const MP4 = 'video/mp4';
+const MOV = 'video/mov';
+const allowedImageTypes = [ 'image/jpeg', 'image/jpg', 'image/png' ];
+const facebookImageTypes = allowedImageTypes.concat( [
+	'image/gif',
+	'image/tiff',
+	'image/tif',
+	'image/bmp',
+] );
+const facebookVideoTypes = [
+	'video/3g2',
+	'video/3gp',
+	'video/3gpp',
+	'video/asf',
+	'video/avi',
+	'video/dat',
+	'video/divx',
+	'video/dv',
+	'video/f4v',
+	'video/flv',
+	'video/gif',
+	'video/m2ts',
+	'video/m4v',
+	'video/mkv',
+	'video/mod',
+	'video/mov',
+	'video/mp4',
+	'video/mpe',
+	'video/mpeg',
+	'video/mpeg4',
+	'video/mpg',
+	'video/mts',
+	'video/nsv',
+	'video/ogm',
+	'video/ogv',
+	'video/qt',
+	'video/tod',
+	'video/ts',
+	'video/vob',
+	'video/wmv',
+];
+
+export const RESTRICTIONS = {
+	twitter: {
+		allowedMediaTypes: allowedImageTypes.concat( [ MP4 ] ),
+		image: {
+			maxSize: 5,
+		},
+		video: {
+			maxSize: 512,
+			maxLength: 140,
+		},
+	},
+	facebook: {
+		allowedMediaTypes: facebookImageTypes.concat( facebookVideoTypes ),
+		image: {
+			maxSize: 4,
+		},
+		video: {
+			maxSize: 10000,
```

Comment:
I'm wondering if we want to set a lower maximum here. I realise Facebook hypothetically supports uploading a 10 GB video, but I'm not sure if that's necessarily something we want to allow, considering the load it might put on our servers and the likelihood of failed uploads.

Reply:
More here: p1672959287715339-slack-C02JJ910CNL

Reply:
Maybe we should leave it as is, and think of it in a followup task?